### DISCUSSION TITLE
How to use forcefield for cofactors in xml format

### DISCUSSION THREAD

**Discussion by JuroDobias** [OpenFreeEnergy/openfe#840] [Q&A]
Hi,
I want to try openfe for my system that contains ATP cofactor. I did MD simulations in amber and gromacs using parameters for ATP obtained from this site http://amber.manchester.ac.uk/. Before that I tried to parametrize ATP with GAFF, but resulting geometries of triphospate were not correct, so I really want to use parameters from mentioned site. I managed to convert .prep and .frcmod file to .xml format by grand.utils.create_ligand_xml function (https://grand.readthedocs.io/en/latest/grand.utils.html#description) and use xml file in openmm simulations. Now I am struggling to use this xml file within openfe. Can you please help me with this issue? It would be also nice to use .xml file for standard ligand parameters.

**Comment by IAlibay** [comment]
@JuroDobias thanks for opening this, we'll triage this internally and try to get back to you as soon as possible. 

You mention struggling to use the XML parameters. Could you let us know what steps you have tried and any associated errors?

**Reply by JuroDobias** [reply]
Thank you for your response.

It seems I managed to do it, but I am not sure if I did not make a mistake.
I noticed that rbfe_settings.forcefield_settings.forcefields contains list of .xml files, so I added my ATP.xml file:
rbfe_settings.forcefield_settings.forcefields.append('inputs/ATP.xml') 
Then I included ATP molecule in protein structure. To do that I had to change input format of protein from pdb to mae:
protein = ProteinComponent(Chem.MaeMolSupplier('inputs/protein_ATP.mae', removeHs=False)[0]). 
If I used pdb I got error:
ValueError: I don't know this Ion or something really went wrong! 	PG	ATP	-603	connectivity0
though the CONNECT statements for ATP were present in pdb file.

Simulation is running without complains. If I do not append ATP.xml to rbfe_settings.forcefield_settings.forcefields, it errored during simulation, so it seems it is getting parameters from my xml file. Are there any issues with this approach?

But anyway, I think it would be better to load protein as pdb and add ATP as SmallMoleculeComponent through sdf file, but I guess for small molecule the forcefield parametrization works differently and I cannot directly specify xml file, can I?
